core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -275,6 +277,7 @@ object KafkaConfig {
   val LogFlushSchedulerIntervalMsProp = "log.flush.scheduler.interval.ms"
   val LogFlushIntervalMsProp = "log.flush.interval.ms"
   val LogFlushOffsetCheckpointIntervalMsProp = "log.flush.offset.checkpoint.interval.ms"
+  val LogFlushStartOffsetCheckpointIntervalMsProp = "log.flush.start.offset.checkpoint.interval.ms"
```

Comment:
I am not sure. It seems safer to have a separate config. The current default value of `LogFlushOffsetCheckpointIntervalMs` is 60 seconds. But we probably want LogFlushStartOffsetCheckpointIntervalMs to be set to a lower value because if a broker fails after user purge data but before the logStartOffset is checkpointed, the offset may not be preserved. Having separate configs allows user to tradeoff between disk write overhead and logStartOffset persistence.

Also, it is probably cheaper to checkpoint logStartOffset than checkpointing recoverOffset. We only need to checkpoint logStartOffset for a partition if its logStartOffset > baseOffset, which only happens when user has explicitly requested data purge. Thus the actually amount of data written to the log-start-offset-checkpoint file is probably much less than the data written to recovery-point-offset-checkpoint.